job-server/src/spark.jobserver/context/StreamingContextFactory.scala

```diff
@@ -0,0 +1,39 @@
+package spark.jobserver.context
+
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.Config
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+import spark.jobserver.{SparkStramingJob, SparkSqlJob, SparkJobBase, ContextLike}
+import spark.jobserver.util.SparkJobUtils
+import spark.jobserver.util.SparkJobUtils._
+
+/**
+ * Created by ZeoS on 3/18/15.
+ */
+class StreamingContextFactory extends SparkContextFactory{
+  import SparkJobUtils._
+
+  type C = StreamingContext with ContextLike
+
+  def makeContext(config: Config, contextConfig: Config, contextName: String): C = {
+    val conf = configToSparkConf(config, contextConfig, contextName)
+    val interval :Int = {
+      if (contextConfig.hasPath("streaming.batch_interval")) {
```

Comment:
The usual paradigm for using the config is to define defaults in reference/application.conf.  That way, `config.getInt("streaming.batch_interval")` is guaranteed to succeed, and always grabs the default value as a default.  Then you don't need the nested if/ etc.